native/cocos/bindings/jswrapper/MappingUtils.h

```diff
@@ -36,33 +37,123 @@ class Object;
+    }
 
+    /**
+     * @brief Check if the key exists in the map
+     */
     template <typename T>
-    static std::enable_if_t<!std::is_void_v<T>, Map::iterator> find(T *v) {
-        if constexpr (std::is_const_v<T>) {
-            return find(reinterpret_cast<void *>(const_cast<std::remove_const_t<T> *>(v)));
+    static bool contains(T *nativeObj) {
+        if constexpr (std::is_void_v<T>) {
+            return __nativePtrToObjectMap->count(nativeObj) > 0;
         } else {
-            return find(reinterpret_cast<void *>(v));
+            auto *kls = JSBClassType::findClass(nativeObj);
+            auto range = __nativePtrToObjectMap->equal_range(nativeObj);
+            for (auto itr = range.first; itr != range.second; itr++) {
+                if (itr->second->_getClass() == kls) {
+                    return true;
+                }
+            }
+            return false;
         }
     }
 
+    /**
+     * @brief Iterate se::Object with specified se::Class
+     */
+    template <typename T, typename Fn>
+    static void forEach(T *nativeObj, const Fn &func) {
+        se::Class *kls = nullptr;
+        if constexpr (!std::is_void_v<T>) {
+            kls = JSBClassType::findClass(nativeObj);
+        }
+        auto range = __nativePtrToObjectMap->equal_range(nativeObj);
+        for (auto itr = range.first; itr != range.second; itr++) {
+            if (kls != nullptr && kls != itr->second->_getClass()) {
+                continue;
+            }
+            func(itr->second);
+        }
+    }
+    /**
+     * @brief Iterate se::Object with specified se::Class
+     */
+    template <typename T, typename Fn1, typename Fn2>
+    static void forEach(T *nativeObj, se::Class *kls, Fn1 eachCallback, Fn2 emptyCallback) {
+        int eleCount = 0;
+        auto range = __nativePtrToObjectMap->equal_range(const_cast<std::remove_const_t<T> *>(nativeObj));
+
+        if (range.first == range.second) { // empty
+            if constexpr (std::is_invocable<Fn2>::value) {
+                emptyCallback();
+            }
+        } else {
+            for (auto itr = range.first; itr != range.second; ++itr) {
+                if (kls != nullptr && kls != itr->second->_getClass()) {
+                    continue;
+                }
+                eleCount++;
+                assert(eleCount < 2);
+                eachCallback(itr->second);
+            }
+            if (eleCount == 0) {
+                if constexpr (std::is_invocable<Fn2>::value) {
+                    emptyCallback();
+                }
+            }
+        }
+    }
+    /**
+     * @brief Call `foundCb` with the filtered se::Object*
+     * 
+     * @tparam T 
+     * @tparam Fn1 
+     * @param nativeObj 
+     * @param kls 
+     * @param foundCb 
+     */
+    template <typename T, typename Fn1>
+    static void with(T *nativeObj, se::Class *kls, Fn1 foundCb) {
+        forEach(nativeObj, kls, foundCb, nullptr);
```

Comment:
The difference between `with` and `forEach` is whether the `emptyCallback` is nullptr?
If so, is it ok that just call it `forEach` as well?